src/components/WidgetPropertiesForm/index.tsx

```diff
@@ -72,23 +81,73 @@ export default function WidgetPropertiesForm({ id, onClose }: WidgetPropertiesFo
     },
   });
 
-  const schema = useMemo(() => {
-    const translatedSchema =
-      verna.configSchema?.properties?.[templateName] || ({} as VernaJSONSchemaType);
+  useEffect(() => {
+    setTranslatedSchema(translateSchema());
+  }, [verna.locale]);
+
+  function translateSchema() {
+    // Get the schema corresponding to the configuration of this widget and clone its
+    const widgetSchema = _.cloneDeep(verna.configSchema?.properties?.[templateName]);
 
     const isMessageKey = (key: PropertyKey): key is keyof typeof messages => {
       return key in messages;
     };
 
-    if (translatedSchema.properties) {
-      for (const [key, value] of Object.entries(translatedSchema.properties)) {
-        if (isMessageKey(key)) {
-          value.title = formatMessage(messages[key]);
-        }
+    // Translate parameters names
+    for (const [key, value] of Object.entries(widgetSchema?.properties || {})) {
+      if (isMessageKey(key)) {
+        value.title = formatMessage({
+          defaultMessage:
+            (verna.locale && verna.schemaTranslations[verna.locale][key]) ||
+            messages[key].defaultMessage,
+          id: messages[key].id,
+        });
       }
     }
-    return translatedSchema;
-  }, [verna.configSchema?.properties?.[templateName]]);
+    return widgetSchema || verna.configSchema || {};
+  }
+
+  function translateEnum(enumValues: JSONSchema7Type[]) {
+    return enumValues.map((value) =>
+      formatMessage({ defaultMessage: DEFAULT_PROPERTY_TRANSLATION, id: value as string }),
+    );
+  }
+
+  /**
+   * Translate items of a widget
+   * It manage either an array of objects or a unique one
+   * If it's an array it will translate every objects
+   *
+   * @param items one or many with enum values
+   */
+  function translateItems(items: VernaJSONSchemaType | VernaJSONSchemaType[] | undefined) {
```

Comment:
We have a `Maybe` util to improve readability
```suggestion
  function translateItems(items: Maybe<VernaJSONSchemaType | VernaJSONSchemaType[]>) {
```